ui/src/clientLibraries/components/ClientCSharpOverlay.tsx

```diff
@@ -17,155 +29,48 @@ const ClientCSharpOverlay: FunctionComponent<{}> = () => {
-      <pre>
-        <code>
-          using InfluxDB.Client;
-          <br />
-          <br />
-          namespace Examples
-          <br />
-          &#123;
-          <br />
-          &nbsp;&nbsp;public class Examples
-          <br />
-          &nbsp;&nbsp;&#123;
-          <br />
-          &nbsp;&nbsp;&nbsp;&nbsp;public static void Main(string[] args)
-          <br />
-          &nbsp;&nbsp;&nbsp;&nbsp;&#123;
-          <br />
-          &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;var client =
-          InfluxDBClientFactory.Create("basepath", "token".ToCharArray());
-          <br />
-          &nbsp;&nbsp;&nbsp;&nbsp;&#125;
-          <br />
-          &nbsp;&nbsp;&#125;
-          <br />
-          &#125;
-          <br />
-        </code>
-      </pre>
-      <h5>Using the client to execute a query</h5>
-      <pre>
-        <code>
-          const string query = "from(bucket: \"my_bucket\") |> range(start:
-          -1h)";
-          <br />
-          <br />
-          var tables = await client.GetQueryApi().QueryAsync(query,
-          "someorgid");
-        </code>
-      </pre>
+      <CodeSnippet copyText={initializeClientCodeSnippet} label="C# Code" />
       <h5>Writing Data</h5>
       <p>
-        Data could be written using InfluxDB Line Protocol, Data Point or POCO
-      </p>
-      <p>
-        <b>InfluxDB Line Protocol</b>
+        <b>Option 1: Example for writing using InfluxDB Line Protocol</b>
       </p>
-      <pre>
-        <code>
-          const string data = "mem,host=host1 used_percent=23.43234543
-          1556896326";
-          <br />
-          <br />
-          using (var writeApi = client.GetWriteApi())
-          <br />
-          &#123;
-          <br />
-          &nbsp;&nbsp;writeApi.WriteRecord("bucketID", "orgID",
-          WritePrecision.Ns, data);
-          <br />
-          &#125;
-          <br />
-        </code>
-      </pre>
+      <CodeSnippet
+        copyText={writingDataLineProtocolCodeSnippet}
+        label="C# Code"
+      />
       <p>
-        <b>Data Point</b>
+        <b>Option 2: Example for writing using a Data Point</b>
       </p>
-      <pre>
-        <code>
-          var point = PointData
-          <br />
-          &nbsp;&nbsp;.Measurement("mem")
-          <br />
-          &nbsp;&nbsp;.Tag("host", "host1")
-          <br />
-          &nbsp;&nbsp;.Field("used_percent", 23.43234543)
-          <br />
-          &nbsp;&nbsp;.Timestamp(1556896326L, WritePrecision.Ns);
-          <br />
-          <br />
-          using (var writeApi = client.GetWriteApi())
-          <br />
-          &#123;
-          <br />
-          &nbsp;&nbsp;writeApi.WritePoint("bucketID", "orgID", point);
-          <br />
-          &#125;
-          <br />
-        </code>
-      </pre>
+      <CodeSnippet copyText={writingDataDataPointCodeSnippet} label="C# Code" />
       <p>
-        <b>POCO</b>
+        <b>
+          Option 3: Example for writing using a POCO and corresponding Class
```

Comment:
~Option 3: Example for writing using a POCO and corresponding Class~ -> Option 3: Use POCO and corresponding Class to write data